frontend/src/metabase/components/Popover.jsx

```diff
@@ -72,96 +74,181 @@ export default class Popover extends Component {
     noOnClickOutsideWrapper: false,
   };
 
-  _getPopoverElement(isOpen) {
-    if (!this._popoverElement && isOpen) {
-      this._popoverElement = document.createElement("span");
-      this._popoverElement.className = "PopoverContainer";
-      document.body.appendChild(this._popoverElement);
-      this._timer = setInterval(() => {
-        const { width, height } = this._popoverElement.getBoundingClientRect();
-        if (this.state.width !== width || this.state.height !== height) {
-          this.setState({ width, height });
-        }
-      }, 100);
-    }
-    return this._popoverElement;
+  _createPopoverElement() {
+    const el = document.createElement("span");
+    el.className = "PopoverContainer";
+    return el;
   }
 
   componentDidMount() {
-    this._renderPopover(this.props.isOpen);
+    this._updateTetherOptions();
+    this._updateMaxHeight();
   }
 
   componentDidUpdate() {
-    this._renderPopover(this.props.isOpen);
+    this._updateTetherOptions();
+    this._updateMaxHeight();
   }
 
-  componentWillUnmount() {
-    if (this._tether) {
-      this._tether.destroy();
-      delete this._tether;
-    }
-    if (this._popoverElement) {
-      this._renderPopover(false);
-      ReactDOM.unmountComponentAtNode(this._popoverElement);
-      if (this._popoverElement.parentNode) {
-        this._popoverElement.parentNode.removeChild(this._popoverElement);
-      }
-      delete this._popoverElement;
-      clearInterval(this._timer);
-      delete this._timer;
+  _updateMaxHeight() {
+    const maxHeight = this._getMaxHeight();
+    if (this.state.maxHeight !== maxHeight) {
+      this.setState({
+        maxHeight,
+      });
     }
   }
 
+  componentWillUnmount() {
+    document.body.removeChild(this._popoverElement);
+    this._tether && this._tether.destroy();
+  }
+
   handleDismissal(...args) {
     if (this.props.onClose) {
       this.props.onClose(...args);
     }
   }
 
-  _popoverComponent() {
+  _buildPopover() {
+    const {
+      id,
+      hasBackground,
+      autoWidth,
+      hasArrow,
+      isOpen,
+      className,
+      style,
+      children,
+      noOnClickOutsideWrapper,
+      dismissOnEscape,
+      dismissOnClickOutside,
+    } = this.props;
     const childProps = {
-      maxHeight: this._getMaxHeight(),
```

Comment:
`_getMaxHeight` calls `ReactDOM.findDOMNode` which can't be called as part of `render`, so I moved the setting of `maxHeight` to state and update it on init/update. This should mirror what it was doing previously.